src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -249,18 +255,31 @@ private boolean areTotalsEqual(InvoiceLine invoiceLine1, InvoiceLine invoiceLine
    * 4. Update corresponding Invoice
    * @param id invoiceLine id to be deleted
    */
-  public CompletableFuture<Void> deleteInvoiceLine(String id) {
-    String query = QUERY_PARAM_START_WITH + id;
-
-    return getInvoices(query, httpClient, ctx, okapiHeaders, logger, lang)
-      .thenApply(invoiceCollection -> invoiceCollection.getInvoices()
-        .get(0))
+  public CompletableFuture<Void> deleteInvoiceLine(String lineId) {
+    return getInvoicesIfExists(lineId).thenApply(invoiceCollection -> invoiceCollection.getInvoices()
+      .get(0))
       .thenCompose(invoice -> protectionHelper.isOperationRestricted(invoice.getAcqUnitIds(), DELETE)
         .thenApply(vvoid -> invoice))
-      .thenCompose(invoice -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, id, lang), httpClient, ctx, okapiHeaders, logger)
+      .thenCompose(invoice -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, lineId, lang), httpClient, ctx, okapiHeaders, logger)
         .thenRun(() -> updateInvoiceAndLinesAsync(invoice)));
   }
 
+  private CompletableFuture<InvoiceCollection> getInvoicesIfExists(String lineId) {
+    String query = QUERY_PARAM_START_WITH + lineId;
+    return getInvoices(query, httpClient, ctx, okapiHeaders, logger, lang).exceptionally(t -> {
+      Throwable cause = t.getCause();
+      // When specified invoice line does not exist
+      if (cause instanceof HttpException && ((HttpException) cause).getCode() == Response.Status.NOT_FOUND.getStatusCode()) {
```

Comment:
If there is no invoice meeting the query criteria, `InvoiceCollection` with empty `invoices` list will be returned.

Reply:
I believe instead of `exceptionally`, `thenApply` method should be used. The method should check if the collection contains one invoice and extracts it (i.e. I believe the method should be `...CompletableFuture<Invoice> getInvoiceIfExists(String lineId)`. Otherwise, `HttpException` has to be thrown